sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -294,23 +254,30 @@ public int getNumShards() {
 
     /**
      * When a sink has requested windowed or triggered output, this method will be invoked to return
-     * the filename. The {@link WindowedContext} object gives access to the window and pane, as
-     * well as sharding information. The policy must return unique and consistent filenames
+     * the file {@link ResourceId resource} to be created given the base output directory and an
+     * optional extension from {@link FileBasedSink} configuration (e.g., {@link CompressionType}).
```

Comment:
In what sense is the extension optional: can it be null? if yes, mark nullable; if no, clarify